LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabClient.java

```diff
@@ -0,0 +1,188 @@
+
+        final List<LabAccount> labAccounts = new ArrayList<>(configInfos.size());
+
+        for (final ConfigInfo configInfo : configInfos) {
+            labAccounts.add(createAccount(configInfo));
+        }
+
+        return labAccounts;
+    }
+
+    private LabAccount createAccount(@NonNull final ConfigInfo configInfo) throws LabApiException {
+        // for guest accounts the UPN is located under homeUpn field
+        String username = configInfo.getUserInfo().getHomeUPN();
+        if (username == null || username.equals("") || username.equalsIgnoreCase("None")) {
+            // for accounts that are NOT guest..the UPN is directly on the UPN field
+            username = configInfo.getUserInfo().getUpn();
+        }
+
+        final String password = getPassword(configInfo);
+
+        return new LabAccount(username, password, UserType.fromName(
+                configInfo.getUserInfo().getUserType())
+        );
+    }
+
+    private List<ConfigInfo> fetchConfigsFromLab(@NonNull final LabQuery query) throws LabApiException {
+        Configuration.getDefaultApiClient().setAccessToken(
+                mLabApiAuthenticationClient.getAccessToken()
+        );
+        try {
+            final ConfigApi api = new ConfigApi();
+            return api.apiConfigGet(
+                    valueOf(query.getUserType()),
+                    valueOf(query.getUserRole()),
+                    valueOf(query.getMfa()),
+                    valueOf(query.getProtectionPolicy()),
+                    valueOf(query.getHomeDomain()),
+                    valueOf(query.getHomeUpn()),
+                    valueOf(query.getB2cProvider()),
+                    valueOf(query.getFederationProvider()),
+                    valueOf(query.getAzureEnvironment()),
+                    valueOf(query.getGuestHomeAzureEnvironment()),
+                    valueOf(query.getAppType()),
+                    valueOf(query.getPublicClient()),
+                    valueOf(query.getSignInAudience()),
+                    valueOf(query.getGuestHomedIn()),
+                    valueOf(query.getHasAltId()),
+                    valueOf(query.getAltIdSource()),
+                    valueOf(query.getAltIdType()),
+                    valueOf(query.getPasswordPolicyValidityPeriod()),
+                    valueOf(query.getPasswordPolicyNotificationDays()),
+                    valueOf(query.getTokenLifetimePolicy()),
+                    valueOf(query.getTokenType()),
+                    valueOf(query.getTokenLifetime())
+            );
+        } catch (final com.microsoft.identity.internal.test.labapi.ApiException ex) {
+            throw new LabApiException(LabError.FAILED_TO_GET_ACCOUNT_FROM_LAB, ex);
+        }
+    }
+
+    private String valueOf(final Object obj) {
+        return obj == null ? null : obj.toString();
+    }
+
+    @Override
+    public LabAccount createTempUser(@NonNull final TempUserType tempUserType) throws LabApiException {
```

Comment:
It would be good if the lab team had called this create TempAccount... it's hard to create a temporary human agent... and a temporary software agent for that matter... nit.

Reply:
They called it `createTempUser` but we can change the name on our end to call the API createTempAccount

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1406/commits/de5b2db2c4cb4c4caad3555a12237704ca9ce6b5